### DISCUSSION TITLE
Road traffic tutorial model 01  - cannot load shapefiles

### DISCUSSION THREAD

**Discussion by kmamor** [gama-platform/gama#168] [Q&A] [answered]
I have worked through the predator/prey tutorial and made a start on Road Traffic. I download the various files from here: https://github.com/gama-platform/gama.old/tree/GAMA_1.9.2/msi.gama.models/models/Tutorials/Road%20Traffic/includes - this includes 12 files 3 x shp, 3 x dbf , 3 x prj and 3 x shx - to the includes folder set up when I stared the new project. 

Even using the code provided I keep getting an error message and the gis data does not load. The error message appears when I try to run the experiment; '2 occurrences in 2 agents at cycle 0: Java error: runtime exception....'

The cursor sits on the code starting 'geometry shape...'

The code I've used (giving up on my own version) is download from github and is: 

```
/**
* Name: Loading of GIS data (buildings and roads)
* Author:
* Description: first part of the tutorial: Road Traffic
* Tags: gis
*/

model tutorial_gis_city_traffic

global {
	file shape_file_buildings <- file("../includes/building.shp");
	file shape_file_roads <- file("../includes/road.shp");
	file shape_file_bounds <- file("../includes/bounds.shp");
	geometry shape <- envelope(shape_file_bounds);
	float step <- 10 #mn;
	
	init {
		create building from: shape_file_buildings with: [type::string(read ("NATURE"))] {
			if type="Industrial" {
				color <- #blue ;
			}
		}
		create road from: shape_file_roads ;
	}
}

species building {
	string type; 
	rgb color <- #gray  ;
	
	aspect base {
		draw shape color: color ;
	}
}

species road  {
	rgb color <- #black ;
	aspect base {
		draw shape color: color ;
	}
}

experiment road_traffic type: gui {
	parameter "Shapefile for the buildings:" var: shape_file_buildings category: "GIS" ;
	parameter "Shapefile for the roads:" var: shape_file_roads category: "GIS" ;
	parameter "Shapefile for the bounds:" var: shape_file_bounds category: "GIS" ;
		
	output {
		display city_display type:3d {
			species building aspect: base ;
			species road aspect: base ;
		}
	}
}
`````
Any help appreciated.

K

**Comment by kmamor** [comment, marked as answer]
Since posting I have dragged the files packaged in with GAMA (form the tutorials folders)  into my project's includes file and now the experiment works fine.  

I also tried loading the files downloaded from GASA's github page into QGIS and they also showed an error. Is it possible the files provided by GAMA are corrupted?

**Comment by lesquoyb** [comment]
Hello @kmamor,
It is possible though highly improbable that the shapefiles from the tutorials are not working. 
Those files and tutorials are also packaged within gama in the model library, could you run those models and tell me if everything works ?
If it's the case I suggest that you copy one whole model from the library and paste it and start fresh from there. If it's not the case then could you upload the whole model directory into a zip file and give us more information about your setup (operating system, gama version etc.)?
ping to @ptaillandier who's our shapefile expert

**Reply by kmamor** [reply]
Yes - per my second post, the shapefiles packaged within the GAMA software (within tutorials folder) work if I copy them over to my project folder. So my query has been resolved. 

However the original shapefiles copied from the GAMA github site (see link in first post) do not work - I cannot open them either in GAMA or QGIS. 

Thanks,

Kevin

**Reply by lesquoyb** [reply]
Ah yes sorry for the misunderstanding.
I've downloaded the 12 files and drag-and-dropped them into my includes folder in a new project, created a model with the code you provided and I have no issues, the model is running fine and showing some buildings and roads.
Is it possible that there was a problem when you downloaded and that's when they became corrupt ?

**Reply by kmamor** [reply]
Hi ,
This is odd. I tried downloading the files again (agreeing that it was probably a corruption issue) and I get the same errors as in the fist post. I then, again, replace these with those copied from the tutorial folder within GAMA and the experiment runs no problem. 

As I have the tutorial working I'm not too worried about pursuing this . 

Thanks for your help,

Kevin

**Reply by kmamor** [reply]
I have moved onto the 'incremental model' tutorial and am having exactly the same problem. When I download the files from [here](https://github.com/gama-platform/gama.old/tree/GAMA_1.9.3/msi.gama.models/models/Tutorials/Incremental%20Model/includes) the model will not run - I get the same error message as my first post.  However if I use the files from the tutorial's includes folder then no problem. 

Is it the way I'm downloading? On the github page I right-click each file and press 'save link as..'. Is there another way of downloading? I cannot find a way of zipping them and downloading, for example. 

K

**Comment by AlexisDrogoul** [comment]
Hi. It is more than probably the way you download individual files that causes this issue. I would strongly recommend to use git to download / maintain these files, which would also allow you to remain up to date. If you're reluctant (like me ^^) in using the command line version, you can have a try at [GitHub Desktop](https://desktop.github.com), which works really well on all platforms.   Once installed, you go back to the root of the repository (in your case https://github.com/gama-platform/gama.old/tree/GAMA_1.9.2) and, in the dropdown menu `Code` select "Open with GitHub Desktop".

**Reply by kmamor** [reply]
Hi Alexis,

I tried that and it worked! Many thanks. The lesson here (for me) is that you cannot download shapefiles one at a time - the linkage to the sfx, prj files etc is broken. I think also from the 'code' button I could have downloaded the repository as a zip file and achieved the same result.  

Problem solved - thanks again.

K
